File: LeetCode/solutions/1_Two_Sum/solution.cpp
#include <stdlib.h>
#include <vector>

using namespace std;

class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        unordered_map<int, int> basket;
        int size = nums.size();

        for (int i = 0; i < size; i++)
        {
            int remain = target - nums[i];

            if (basket.count(remain))
            {
                return {basket[remain], i};
            }

            basket[nums[i]] = i;
        }

        return {};
    }
};

File: LeetCode/solutions/1_Two_Sum/solution.dart
class Solution {
  List<int> twoSum(List<int> nums, int target) {
    Map<int, int> basket = {};

    for (int i = 0; i < nums.length; i++) {
      int remain = target - nums[i];

      if (basket.containsKey(remain)) {
        return [basket[remain]!, i];
      }

      basket[nums[i]] = i;
    }

    return [];
  }
}
